apps/sharedtest/test/unit/advanced_telemetry_helper_test.js

```diff
@@ -0,0 +1,225 @@
+      ath.count('mymetric');
+      sinon.assert.calledWith(console.info, 'telemetry|mymetric|1|4');
+      done();
+    });
+
+    test('should not write format the message if missing param',
+    function(done) {
+      ath.count();
+
+      sinon.assert.notCalled(console.info);
+      // Should call a warning message.
+      sinon.assert.calledOnce(console.warn);
+      done();
+    });
+  });
+
+  suite('linear metric formatting:', function() {
+    var ath;
+
+    setup(function(done) {
+      this.sinon.stub(console, 'info');
+      // Suppress the error message
+      this.sinon.stub(console, 'warn');
+
+      ath = new AdvancedTelemetryHelper();
+      MockNavigatorSettings.mTriggerObservers(
+        AdvancedTelemetryHelper.TELEMETRY_ENABLED_KEY, { settingValue: true });
+      done();
+    });
+
+    teardown(function() {
+    });
+
+    test('should call console info once', function(done) {
+      ath.linear('mymetric', 10, 1, 10000, 5);
+      sinon.assert.calledOnce(console.info);
+      done();
+    });
+
+    test('should format the message correctly with full params',
+    function(done) {
+      ath.linear('mymetric', 10, 0, 10000, 5);
+      sinon.assert.calledWith(console.info,
+        'telemetry|mymetric|10|1|0|10000|5');
+      done();
+    });
+
+    test('should format the message correctly without full params',
+    function(done) {
+      ath.linear('mymetric', 10);
+      sinon.assert.calledWith(console.info, 'telemetry|mymetric|10|1|');
+      done();
+    });
+
+    test('should not write format message if missing param', function(done) {
+      ath.linear();
+      sinon.assert.notCalled(console.info);
+      // Should call a warning message.
+      sinon.assert.calledOnce(console.warn);
+      done();
+    });
+  });
+
+  suite('exponential metric formatting:', function() {
+    var ath;
+
+    setup(function(done) {
+      this.sinon.stub(console, 'info');
+      // Suppress the error message
+      this.sinon.stub(console, 'warn');
+
+      ath = new AdvancedTelemetryHelper();
+      MockNavigatorSettings.mTriggerObservers(
+        AdvancedTelemetryHelper.TELEMETRY_ENABLED_KEY, { settingValue: true });
+      done();
+    });
+
+    teardown(function() {
+    });
+
+    test('should call console info once', function(done) {
+      ath.exp('mymetric', 10, 0, 10000, 5);
+      sinon.assert.calledOnce(console.info);
+      done();
+    });
+
+    test('should format the message correctly with full params',
+    function(done) {
+      ath.exp('mymetric', 10, 1, 10000, 5);
+      sinon.assert.calledWith(console.info,
+        'telemetry|mymetric|10|0|1|10000|5');
+      done();
+    });
+
+    test('should format message correctly without full params',
+    function(done) {
+      ath.exp('mymetric', 10);
+      sinon.assert.calledWith(console.info,
+        'telemetry|mymetric|10|0|');
+      done();
+    });
+
+    test('should not write format the message if missing param',
+    function(done) {
+      ath.exp();
```

Comment:
I think it's worth having additional negative test cases where, for example, there are three and four arguments specified.